### PR TITLE
APB-704-additional-tests

### PR DESCRIPTION
**Pull request by ralphcollett**

### CODE REVIEW

File: src/test/scala/uk/gov/hmrc/agentmtdidentifiers/model/UtrSpec.scala [hunk 1]
```diff
@@ -36,11 +47,15 @@ class UtrSpec extends FlatSpec with Matchers {
     Utr.isValid("200000000B") shouldBe false
   }
 
+  it should "be false when it us akk non-digit characters" in {
```
Comment (frankoid) [inline src/test/scala/uk/gov/hmrc/agentmtdidentifiers/model/UtrSpec.scala]: What value does this add given the "be false when it has non-digit characters" case above?

File: src/test/scala/uk/gov/hmrc/agentmtdidentifiers/model/UtrSpec.scala [hunk 2]
```diff
@@ -22,12 +22,23 @@ class UtrSpec extends FlatSpec with Matchers {
...
 
   it should "be false when it has more than 10 digits" in {
     Utr.isValid("20000000000") shouldBe false
   }
 
+  it should "be false when it is empty" in {
+    Utr.isValid("") shouldBe false
+  }
+
+  it should "be false when it is all zeros" in {
```
Comment (frankoid) [inline src/test/scala/uk/gov/hmrc/agentmtdidentifiers/model/UtrSpec.scala]: Why is it important for an all-zero UTR to be considered invalid?
Is this just a special case of "checksum fails"?

File: src/test/scala/uk/gov/hmrc/agentmtdidentifiers/model/UtrSpec.scala (hunk 1 above)
Comment (ralphcollett) [inline src/test/scala/uk/gov/hmrc/agentmtdidentifiers/model/UtrSpec.scala]: True, I am happy to remove if you are happy @sarthak83? Spelling could do with some work too.

File: src/test/scala/uk/gov/hmrc/agentmtdidentifiers/model/UtrSpec.scala (hunk 2 above)
Comment (ralphcollett) [inline src/test/scala/uk/gov/hmrc/agentmtdidentifiers/model/UtrSpec.scala]: I agree too in terms of test coverage. I was following the cases in the Jira. @sarthak83?

File: src/test/scala/uk/gov/hmrc/agentmtdidentifiers/model/UtrSpec.scala (hunk 1 above)
Comment (sarthak83) [inline src/test/scala/uk/gov/hmrc/agentmtdidentifiers/model/UtrSpec.scala]: Yeah I am happy for it to be removed @ralphcollett

File: src/test/scala/uk/gov/hmrc/agentmtdidentifiers/model/UtrSpec.scala (hunk 2 above)
Comment (sarthak83) [inline src/test/scala/uk/gov/hmrc/agentmtdidentifiers/model/UtrSpec.scala]: @frankoid yes, it was intended as a special checksum fail scenario as first digit can never be 0. Happy either ways, to keep it, or remove it.

**Comment by ralphcollett** [comment]
@frankoid and @sarthak83 - note that excessive test cases have now been removed. Thank-you for the feedback.

**Pull request merged**